### DISCUSSION TITLE
Photo Sphere Viewer 5 is available for preview!

### DISCUSSION THREAD

**Discussion by mistic100** [mistic100/Photo-Sphere-Viewer#790] [Announcements]
I am pleased to announce that the next major version of Photo Sphere Viewer is available for testing right now.

The main changes are:
- the NPM package has been splitted into multiple packages for the core and each adapter and plugin
- the library now uses modern ES6 syntax published in two formats : UMD & ESM
- the notoriously badly named `longitude` and `latitude` parameters has been renamed to `yaw` and `pitch`
- uEvent library has been dropped to use native events

You can test this version right now by installing `@photo-sphere-viewer/core@alpha` and follow the [migration guide](https://photo-sphere-viewer-5.netlify.app/guide/migration.html). Also check the [API documentation](https://photo-sphere-viewer-5.netlify.app/api/) (the header link points to v4).

Please comment here if you have any remarks or bug report specific to this version (no feature requests).

**Comment by Elius94** [comment]
Great! as soon as it is final I will update the React wrapper

**Comment by mistic100** [comment]
First "bug" : TypeScript is configured to target "esnext", which uses expetimental features like class properties, will change the target to "es2021"

**Comment by haldo98** [comment]
> * the notoriously badly named `longitude` and `longitude` parameters has been renamed to `yaw` and `pitch`

There's a typo in the explanation... longitude twice....

**Comment by haldo98** [comment]
I tried the new v5
https://www.tadini.it/panorama/andolla/panorama_andolla.html    ( html is for v5 .. while htm is still for v4 )

I get two errors:

psv.ts:27 PhotoSphereViewer: Unknown option defaultYaw

J @ psv.ts:27
Viewer.ts:120 Uncaught TypeError: Cannot read properties of undefined (reading 'adapter')
    at new Ni (Viewer.ts:120:40)
    at panorama_andolla.html:58:18


if  I delete the defaultYaw line the pano loads but other error on compass and marker plugins  appears

this is the page without defaultYaw
https://www.tadini.it/panorama/andolla/panorama_andolla2.html:

Hope it helps

**Reply by mistic100** [reply]
You are not using the "alpha" version of the libs, for example https://cdn.jsdelivr.net/npm/@photo-sphere-viewer/core@alpha/index.min.js

So you are getting an earlier build that contains errors.

But I will release the final 5.0.0 tomorrow.
